flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -339,15 +348,29 @@ public Builder setSnapshotProperty(String property, String value) {
      */
     @Deprecated
     public DataStreamSink<RowData> build() {
-      return chainIcebergOperators();
+      return append();
     }
 
     /**
      * Append the iceberg sink operators to write records to iceberg table.
      *
      * @return {@link DataStreamSink} for sink.
      */
-    public DataStreamSink<Void> append() {
+    @SuppressWarnings("rawtypes")
```

Comment:
Why change this public method's return data type ?  Be careful it's a incompatibility changes.  In the first version, w e introduced the `FlinkSink#build` method with an incorrect return data type, and then we marked it as deprecated in this [PR](https://github.com/apache/iceberg/pull/3061)

If any changes are planning to make in this PR, then we will need to follow the same approach. In my thought, I don't think we need to change this,  the current `DataStreamSink<Void>` is the correct return type.

Reply:
Reference here:https://github.com/apache/iceberg/pull/3998#discussion_r812534842

Reply:
It's okay to suppress unchecked, I didn't mean to imply in that comment that the use of raw types is okay. This should cast, not remove type parameters.